Nodejs/Product/Nodejs/SharedProject/HierarchyIdMap.cs

```diff
@@ -29,61 +29,37 @@ public uint Add(HierarchyNode node)
                     }
                 }
             }
-#endif
 
-            VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread();
-
-            lock (this.theLock)
+#endif
+            if (!this.freedIds.TryPop(out var idx))
             {
-                if (this.freedIds.Count > 0)
-                {
-                    var i = this.freedIds.Pop();
-                    this.ids[i] = new WeakReference<HierarchyNode>(node);
-                    return (uint)i + 1;
-                }
-                else
-                {
-                    this.ids.Add(new WeakReference<HierarchyNode>(node));
-                    // ids are 1 based
-                    return (uint)this.ids.Count;
-                }
+                idx = this.nodes.Count;
             }
+
+            var addSuccess = this.nodes.TryAdd(idx, new WeakReference<HierarchyNode>(node));
+            Debug.Assert(addSuccess, "Failed to add a new item");
+
+            return (uint)idx + 1;
```

Comment:
>+ 1 [](start = 29, length = 3)

Given that we're no longer indexing into a list, is there any value in maintaining off-by-one indices?